From d8ad62f241c522e284b818f63cbb3d4414baf961 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 8 Feb 2022 01:37:34 +1100
Subject: [PATCH 3/3] remove build-system from pyproject.toml

---
 pyproject.toml | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index e28ffe5..e43058a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,7 +19,3 @@ exclude = '''
 [tool.isort]
 profile = "black"
 multi_line_output = 3
-
-[build-system]
-requires = ["setuptools", "wheel"]
-build-backend = "setuptools.build_meta"
-- 
2.32.0.windows.2

